From 5bea4346f35478108fa7c5ce8ade4dccede9775a Mon Sep 17 00:00:00 2001
From: "deepak.s" <deepak.s@adlinktech.com>
Date: Tue, 17 Mar 2026 16:28:27 +0000
Subject: [PATCH] final

---
 board/mediatek/mt8370/mt8370_evk.c | 85 ++++++++++++++++++++++++++++++
 1 file changed, 85 insertions(+)

diff --git a/board/mediatek/mt8370/mt8370_evk.c b/board/mediatek/mt8370/mt8370_evk.c
index 889e8dce6a..d269fd804d 100644
--- a/board/mediatek/mt8370/mt8370_evk.c
+++ b/board/mediatek/mt8370/mt8370_evk.c
@@ -12,6 +12,7 @@
 #include <linux/kernel.h>
 #include <linux/arm-smccc.h>
 #include <iot_ab.h>
+#include <init.h>
 
 #define MT8370_UPDATABLE_IMAGES	5
 
@@ -124,3 +125,87 @@ int board_init(void)
 
 	return 0;
 }
+
+#define SKU_REG 0x10005230
+
+#define SKU_BIT0 11
+#define SKU_BIT1 12
+#define SKU_BIT2 22
+#define SKU_BIT3 16
+#define SKU_BIT4 17
+
+struct sku_info {
+    uint32_t sku_id;
+    char* sku_str;
+       const char* dtbo_file;
+};
+
+struct sku_info skus[] = {
+    {0, "OSM-MTK510-4G-32G-ER(000E)", "conf-temp-er.dtbo#conf-memory-4G.dtbo"},
+    {2, "OSM-MTK510-8G-32G-ER/PSI (010E)", "conf-temp-er.dtbo#conf-memory-8G.dtbo"},
+    {4, "OSM-MTK510-2G-32G-CT/NVM(110E)", "conf-memory-2G.dtbo"},
+    {20, "OSM-MTK510-2G-32G-CT(100E)", "conf-memory-2G.dtbo"}
+};
+
+#define PCB_REG                0x10005240
+#define PCB_BIT0       21
+#define PCB_BIT1       22
+
+
+
+
+int board_late_init(void)
+{
+       char *boot_conf_orig = env_get("boot_conf");
+       char cmd[256];
+
+    uint32_t sku_id_0 = (*(volatile uint32_t *)SKU_REG >> SKU_BIT0) & 0x1;
+    uint32_t sku_id_1 = (*(volatile uint32_t *)SKU_REG >> SKU_BIT1) & 0x1;
+    uint32_t sku_id_2 = (*(volatile uint32_t *)SKU_REG >> SKU_BIT2) & 0x1;
+    uint32_t sku_id_3 = (*(volatile uint32_t *)SKU_REG >> SKU_BIT3) & 0x1;
+    uint32_t sku_id_4 = (*(volatile uint32_t *)SKU_REG >> SKU_BIT4) & 0x1;
+
+    uint32_t sku_id = (sku_id_4 << 4) | (sku_id_3 << 3) | (sku_id_2 << 2) | (sku_id_1 << 1) | sku_id_0;
+
+    int i;
+    for (i = 0; i < sizeof(skus)/sizeof(struct sku_info); i++) {
+        if (skus[i].sku_id == sku_id) {
+            printf("SKU:   %s\n", skus[i].sku_str);
+                       snprintf(cmd, sizeof(cmd), "setenv boot_conf '%s#%s'", boot_conf_orig, skus[i].dtbo_file);
+                       run_command(cmd, 0);
+            break;
+        }
+    }
+
+    if (i == sizeof(skus)/sizeof(struct sku_info)) {
+        printf("Unknown SKU\n");
+    }
+
+    uint32_t reg_value;
+
+    uint32_t pcb_id_0 = (*(volatile uint32_t *)PCB_REG >> PCB_BIT0) & 0x1;
+    uint32_t pcb_id_1 = (*(volatile uint32_t *)PCB_REG >> PCB_BIT1) & 0x1;
+
+    uint32_t pcb_id = (pcb_id_1 << 1) | pcb_id_0;
+
+    switch (pcb_id) {
+    case 0:
+        printf("PCB:   PCB A1\n");
+        break;
+    case 1:
+        printf("PCB:   PCB A2\n");
+        break;
+    case 2:
+        printf("PCB:   PCB A3\n");
+        break;
+    case 3:
+        printf("PCB:   PCB A4\n");
+        break;
+    default:
+        printf("Unknown PCB\n");
+        break;
+    }
+
+    return 0;
+}
+
-- 
2.43.0

