From 7d66a76b608e1945a6363188afe0b1320e2fd70f Mon Sep 17 00:00:00 2001
From: "deepak.s" <deepak.s@adlinktech.com>
Date: Tue, 27 Aug 2024 11:46:15 +0530
Subject: [PATCH] Added DSI panel support

---
 drivers/video/mediatek/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/mediatek/Makefile b/drivers/video/mediatek/Makefile
index cdf12497d0..0782280031 100644
--- a/drivers/video/mediatek/Makefile
+++ b/drivers/video/mediatek/Makefile
@@ -22,7 +22,7 @@ ifeq ($(CONFIG_VIDEO_MEDIATEK),y)
 			dp_mt8188.o \
 			vdosys_mt8188.o \
 			disp_pwm_mt8188.o
-		obj-$(CONFIG_TARGET_MT8195) += \
+		obj-$(CONFIG_TARGET_LEC_MTK_I1200) += \
 			dsi_mt8195.o \
 			dp_mt8195.o \
 			vdosys_mt8195.o \
-- 
2.45.2

